dnf/dnf-main.c

```diff
@@ -40,21 +41,55 @@ process_global_option (const gchar  *option_name,
   g_autoptr(GError) local_error = NULL;
   DnfContext *ctx = DNF_CONTEXT (data);
 
-  gboolean ret;
+  gboolean ret = TRUE;
   if (g_strcmp0 (option_name, "--disablerepo") == 0)
     {
-      ret = show_help ? TRUE : dnf_context_repo_disable (ctx, value, &local_error);
+      enable_disable_repos = g_slist_append (enable_disable_repos, g_strconcat("d", value, NULL));
     }
   else if (g_strcmp0 (option_name, "--enablerepo") == 0)
     {
-      ret = show_help ? TRUE : dnf_context_repo_enable (ctx, value, &local_error);
+      enable_disable_repos = g_slist_append (enable_disable_repos, g_strconcat("e", value, NULL));
+    }
+  else if (g_strcmp0 (option_name, "--installroot") == 0)
+    {
+      if (value[0] != '/')
+        {
+          local_error = g_error_new (G_OPTION_ERROR,
+                                     G_OPTION_ERROR_BAD_VALUE,
+                                     "Absolute path must be used");
+          ret = FALSE;
+        }
+      else
+        {
+          dnf_context_set_install_root (ctx, value);
+
+          /*
+           * If /etc/yum.repos.d/ or /etc/repos.d/ subdirectory exists in installroot directory,
+           * it will be used.
+           */
+          const char * const REPOS_CFG_DIR_PATHS[] = {"/etc/yum.repos.d/", "/etc/repos.d/", NULL};
```

Comment:
Where did `/etc/repos.d` come from? Is that a path we support somewhere?

Reply:
@Conan-Kudo Oh. You are right. I looked to the history. Dnf is looking for these directories "/etc/yum.repos.d", "/etc/yum/repos.d", "/etc/distro.repos.d". I will fix it.

Reply:
@jrohel We also need vars parsed, right?

Reply:
@Conan-Kudo The basic vars (eg. releasever) are detected from installroot (if not overided by command line). Loading vars from files and environment is not working in microdnf at all. I have to finish some work for it. But I hope --installroot is usable just now.

Reply:
Don't we have an API in libdnf to fetch this stuff now? I vaguely recall that DNF now uses that...

Reply:
Yes, we have API for configuring paths to vars directories. It is not fully functional in libdnf yet. It depends on https://github.com/rpm-software-management/libdnf/pull/691 . I must finish it. I'm stopped the work for a while because I am thinking about DNF API compatibility.
Anyway, idea is, that libdnf will read vars from installroot directory by default. So there is not need for setup from microdnf side.